### PR TITLE
State filtering and cost graph initialization

### PR DESCRIPTION
**Pull request by TJTapajyoti**
Initialize cost graph should be always set to true because the cost graph pickle will not exist for new users. 
Setting this to false should only be done when exploring same regions multiple times on the same machine. 

The code is setup such that data filtering is always true unless user sends 'US' as an argument to the code run statement. 
Sending any other arguments, namely state abbreviations will result in filtering to occur
Send python __main__.py --list US for national scale run
Send python __main__.py --list IO CO .... for state runs. 
maximum of 6 states can be entered on the HPC runs right now. Can be modified later.

### CODE REVIEW

File: celavi/__main__.py
Comment (rjhanes) [inline celavi/__main__.py:99]: I don't think over-writing an input parameter value by default is a good idea. The way the filtering will work when we have the config files up and running is that `data_filtering_choice` and the list of states to filter are both inputs (in the config file), and the filtering code only executes if `data_filtering_choice` is True. I think we should keep it that way, partially to avoid over-writing the `data_filtering_choice` and partially because having to send the 'US' flag will be misleading when running CELAVI on a dataset that doesn't need filtering but isn't national scale.

If this functionality currently works better on HPC, then as a stopgap measure until the config file functionality is merged into `develop` I recommend not merging this PR and continuing to execute HPC runs off this branch. I can prioritize completing the config file functionality and try to get that PR tested and merged into `develop` by tomorrow morning.

**Review by rjhanes — CHANGES_REQUESTED** [review]
See my comment on the code - I think we shouldn't merge this PR because the config functionality will address this issue.

File: celavi/__main__.py
Comment (TJTapajyoti) [reply celavi/__main__.py:99]: > I don't think over-writing an input parameter value by default is a good idea. The way the filtering will work when we have the config files up and running is that `data_filtering_choice` and the list of states to filter are both inputs (in the config file), and the filtering code only executes if `data_filtering_choice` is True. I think we should keep it that way, partially to avoid over-writing the `data_filtering_choice` and partially because having to send the 'US' flag will be misleading when running CELAVI on a dataset that doesn't need filtering but isn't national scale.
> 
> If this functionality currently works better on HPC, then as a stopgap measure until the config file functionality is merged into `develop` I recommend not merging this PR and continuing to execute HPC runs off this branch. I can prioritize completing the config file functionality and try to get that PR tested and merged into `develop` by tomorrow morning.

Agreed.

**Comment by TJTapajyoti** [comment]
@rjhanes You can close this PR as config facility has been incorporated.
For HPC runs, I need to build a yaml overwriter that during the HPC run first runs a python script that overwrites the original config.yaml file with my intended states. Otherwise job submission is not possible. 

That is outside of CELAVI and will not show up over here.

**Pull request closed without merging**
